### PR TITLE
add runit finish option

### PR DESCRIPTION
**Pull request by ptqa**
If someone else needs finish script support here.

### CODE REVIEW

File: metadata.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@
 license           'Apache 2.0'
 description       'Manages client.rb configuration and chef-client service'
 long_description  IO.read(File.join(File.dirname(__FILE__), 'README.md'))
-version           '4.0.0'
+version           '4.0.1'
```
Comment (jtimberman) [inline metadata.rb]: Please don't bump the version in pull requests. The maintainers will handle versioning.

**Comment by jtimberman** [comment]
Aside from the note about versions what is the purpose of this? What's the use case?

**Comment by ptqa** [comment]
@jtimberman with this you can override this attribute and use finish script for removing temporary files, killing hanged up processes, waiting for 5 sec, etc.

**Comment by jtimberman** [comment]
I understand the purpose of a finish script with runit. What is the use case with chef-client? I'm hesitant to have yet another knob to turn (attributive) that can increase the cognitive load for someone trying to implement this cookbook in their environment. In 6 years of running chef-client in runit I've not personally needed to have a finish script - the client behaves pretty well for me. 

Is there some behavior of chef-client that needs to have clean up performed like this? Perhaps there is a bug we should address in chef itself?

**Comment by ptqa** [comment]
@jtimberman well, the problem I faced is chef-client hangs up, but I have not yet managed to find out why. So I run chef-client using timeout util and use finish script for waiting interval time. I understand that it's not a best way to manage it.

**Pull request closed without merging**